packages/meteor/dynamics_nodejs.js

```diff
@@ -60,20 +60,10 @@ class EnvironmentVariableAsync {
     }
 
     return Meteor._runAsync(
-      async function () {
-        let ret;
-        try {
-          Meteor._updateAslStore(CURRENT_VALUE_KEY_NAME, value);
-          Meteor._updateAslStore(UPPER_CALL_DYNAMICS_KEY_NAME, dynamics);
-          ret = await func();
-        } finally {
-          Meteor._updateAslStore(CURRENT_VALUE_KEY_NAME, undefined);
-          Meteor._updateAslStore(
-            UPPER_CALL_DYNAMICS_KEY_NAME,
-            undefined,
-          );
-        }
-        return ret;
+      function () {
+        Meteor._updateAslStore(CURRENT_VALUE_KEY_NAME, value);
+        Meteor._updateAslStore(UPPER_CALL_DYNAMICS_KEY_NAME, dynamics);
+        return func();
```

Comment:
This will not work.

We need to wait for the `func()` to finish and reset the states of  `CURRENT_VALUE_KEY_NAME` and `UPPER_CALL_DYNAMICS_KEY_NAME` after the `func()` is done.

This is because you can call a `withValues` inside another, and you need to set and reset those states to keep context between calls.

Otherwise, you'll get issues like this: https://github.com/meteor/meteor/pull/13063

Reply:
#13063 changed it to create a new dynamics object each time instead of mutating the parent dynamics. This should remove the need to reset the state since the parent and children all have their own state. If you comment out the reset code, the example code from the PR description stills works correctly.

Reply:
I agree with @zodern, I will include that code as a test just to make sure.